olp-cpp-sdk-core/src/client/OlpClient.cpp

```diff
@@ -185,16 +305,61 @@ NetworkAsyncCallback GetRetryCallback(
   };
 }
 
+NetworkAsyncCallback GetNewRetryCallback(
+    int current_try, int current_backdown_period, const RetrySettings& settings,
+    const NetworkAsyncCallback& callback,
+    const std::shared_ptr<http::NetworkRequest>& network_request,
+    std::shared_ptr<http::Network> network,
+    const std::weak_ptr<CancellationContext>& weak_cancel_context) {
+  ++current_try;
+  return [=](HttpResponse response) {
+    if (current_try >= settings.max_attempts ||
+        !settings.retry_condition(response)) {
+      callback(response);
+    } else {
+      // TODO there must be a better way
+      std::this_thread::sleep_for(
+          std::chrono::milliseconds(current_backdown_period));
+
+      auto next_backdown_period =
+          settings.backdown_policy(current_backdown_period);
+      auto cancel_context = weak_cancel_context.lock();
+      if (cancel_context) {
+        cancel_context->ExecuteOrCancelled(
+            [&]() -> CancellationToken {
+              return ExecuteSingleRequest(
+                  network, *network_request,
+                  GetNewRetryCallback(current_try, next_backdown_period,
+                                      settings, callback, network_request,
+                                      network, weak_cancel_context));
+            },
+            [callback]() {
+              callback(HttpResponse(network::Network::ErrorCode::Cancelled,
+                                    "Operation Cancelled."));
+            });
+      } else {
+        callback(HttpResponse(network::Network::ErrorCode::UnknownError,
+                              "Unknown error."));
+      }
+    }
+  };
+}
+
 CancellationToken OlpClient::CallApi(
     const std::string& path, const std::string& method,
-    const std::multimap<std::string, std::string>& queryParams,
-    const std::multimap<std::string, std::string>& headerParams,
-    const std::multimap<std::string, std::string>& /*formParams*/,
-    const std::shared_ptr<std::vector<unsigned char>>& postBody,
-    const std::string& contentType,
+    const std::multimap<std::string, std::string>& query_params,
+    const std::multimap<std::string, std::string>& header_params,
+    const std::multimap<std::string, std::string>& form_params,
+    const std::shared_ptr<std::vector<unsigned char>>& post_body,
+    const std::string& content_type,
     const NetworkAsyncCallback& callback) const {
-  auto networkRequest = CreateRequest(path, method, queryParams, headerParams,
-                                      postBody, contentType);
+  if (settings_.network_request_handler) {
+    return CallNewNetworkApi(path, method, query_params, header_params,
+                             form_params, post_body, content_type, callback);
+  }
+
+  auto network_request = CreateRequest(path, method, query_params,
```

Comment:
Why do we need to keep old Network path? This I still don't get.
You keep the API intact, use the new http::Network and convert right before calling callback from http::HttpResponse to network::HttpResponse, adapt exisiting unit tests.